workitem/expression_compiler.go

```diff
@@ -124,6 +128,24 @@ func (c *expressionCompiler) Equals(e *criteria.EqualsExpression) interface{} {
 	return c.binary(e, "=")
 }
 
+func (c *expressionCompiler) Substring(e *criteria.SubstringExpression) interface{} {
+	if isInJSONContext(e.Left()) {
+		l := e.Left().(*criteria.FieldExpression).FieldName
+		if strings.Contains(l, "'") {
+			// beware of injection, it's a reasonable restriction for field names,
+			// make sure it's not allowed when creating wi types
```

Comment:
What about other parts of SQL injection? What about a semicolon for example?

Reply:
In this particular case, single quote is already opened. So, the remaining text will be treated as field name until another single quote appears. So, even if a semicolon comes, that will be treated as field name. So, I think we are safe here.

Reply:
can you verify it with a test ?

Reply:
> can you verify it with a test ?

@baijum , @xcoulon is write, please validate with a test.

Reply:
Fixed. 675da0ea05afa2576338e2d44c3dcbb916bd3f63